### PR TITLE
Roap state machine new

### PR DESCRIPTION
**Pull request by chrisadubois**
# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Test Coverage

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* SDK Version
* Node/Browser Version
* NPM Version

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/node_modules/@webex/plugin-meetings/src/roap/handler.js [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+          });
+        return true;
+      }
+    }
+    if (!RoapUtil.ensureMeeting(meeting, action.type)) {
+      return true;
+    }
+  }
+  return false;
+};
```
Comment (andrewholsted) [inline packages/node_modules/@webex/plugin-meetings/src/roap/handler.js]: This method is a little hard to read/follow but I don't have any drastic improvements.

File: packages/node_modules/@webex/plugin-meetings/src/roap/handler.js [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+      case ROAP.ROAP_STATE.GLARE:
+        session.GLARE_OFFER.tieBreaker = session.GLARE_OFFER.tieBreaker || 0;
+        session.OFFER.tieBreaker = session.OFFER.tieBreaker || 0;
+        config.console.error('Resolve the GLARE condition');
+        if (session.GLARE_OFFER.tieBreaker < session.OFFER.tieBreaker) {
+          // 2
+          config.console.error('local offer wins');
+        }
+        else {
+          config.console.error('remote offer wins', session);
```
Comment (andrewholsted) [inline packages/node_modules/@webex/plugin-meetings/src/roap/handler.js]: We'll want to change these to use `this.logger.error` but I figure we'll take a big pass of all the files and do that at a later point

File: packages/node_modules/@webex/plugin-meetings/src/roap/handler.js [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    if (session && session.state) {
+      handleSessionStep(action, session);
+      signal = ROAP.ROAP_SIGNAL[`${prefix}${action.msg.messageType}`];
+      session.state.step(signal, meeting, action);
+      this.perform(session, meeting, action);
+    }
+  }
+
+  handleAction(session, action, meeting, correlationId) {
+    let signal;
```
Comment (andrewholsted) [inline packages/node_modules/@webex/plugin-meetings/src/roap/handler.js]: What's the purpose of `signal` here? It loosk like it' sonly used in send/receive roap messages but it's undefined. It's later set but then we don't use it anywhere.

File: packages/node_modules/@webex/plugin-meetings/src/roap/handler.js [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+        RoapCollection.deleteSessionSequence(correlationId, action.msg.seq);
+        console.info('Call-RoapHandler#RESET_ROAP_STATE: resetting the RoapHandler state', action.msg.seq);
+        break;
+      default:
+        return true;
+    }
+    return true;
+  }
+
+  submit(action) {
```
Comment (andrewholsted) [inline packages/node_modules/@webex/plugin-meetings/src/roap/handler.js]: This is nice

File: packages/node_modules/@webex/plugin-meetings/src/roap/util.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+  const screenOffer = meeting.screenPeerConnection && meeting.screenPeerConnection.signalingState === SDP.HAVE_LOCAL_OFFER;
+  if (mediaOffer || screenOffer) {
+    return false;
+  }
+  return true;
+};
+
+RoapUtil.handleError = (pc) => PeerConnectionManager.rollBackLocalDescription({mediaPeerConnection: pc})
+  .then(() => Promise.resolve(true))
+  .catch((err) => Promise.reject(err));
```
Comment (andrewholsted) [inline packages/node_modules/@webex/plugin-meetings/src/roap/util.js]: If we aren't doing any logging or actions, you don't need the `.then` and `.catch` blocks. I guess if you explicitly want to return a `true` boolean on success then you can keep `.then(() => true)`

File: packages/node_modules/@webex/plugin-meetings/src/roap/handler.js (hunk 2 above)
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/src/roap/handler.js]: yes

File: packages/node_modules/@webex/plugin-meetings/src/roap/handler.js (hunk 3 above)
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/src/roap/handler.js]: we will adjust once arun is back, wanted to leave his there

File: packages/node_modules/@webex/plugin-meetings/src/roap/util.js (hunk 5 above)
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/src/roap/util.js]: yeah explicitly want to return true

**Pull request merged**